### PR TITLE
builddep: do not check GPG key of SRPM (RhBug:1431486)

### PR DESCRIPTION
**Pull request by ignatenkobrain**
And really bail on error.

References: https://bugzilla.redhat.com/show_bug.cgi?id=1431486
Reported-by: Zbigniew Jędrzejewski-Szmek <zbyszek@in.waw.pl>
Signed-off-by: Igor Gnatenko <ignatenko@redhat.com>

### CODE REVIEW

**Comment by j-mracek** [comment]
@ignatenkobrain Please can you fix uni-test, because it cannot build dnf-plugins-core
```
ERROR: test_source (tests.test_builddep.BuildDepCommandTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/dnf-plugins-core-1.0.2_1-2g82113c0/tests/test_builddep.py", line 47, in test_source
    support.command_run(self.cmd, [SOURCE])
  File "/builddir/build/BUILD/dnf-plugins-core-1.0.2_1-2g82113c0/tests/support.py", line 45, in command_run
    return cmd.run()
  File "/builddir/build/BUILD/dnf-plugins-core-1.0.2_1-2g82113c0/plugins/builddep.py", line 45, in inner
    return func(*args, **kwds)
  File "/builddir/build/BUILD/dnf-plugins-core-1.0.2_1-2g82113c0/plugins/builddep.py", line 118, in run
    self._src_deps(pkgspec)
  File "/builddir/build/BUILD/dnf-plugins-core-1.0.2_1-2g82113c0/plugins/builddep.py", line 148, in _src_deps
    with os.open(src_fn, os.O_RDONLY) as fd:
AttributeError: __exit__
```

**Comment by ignatenkobrain** [comment]
@j-mracek should be fixed now

File: plugins/builddep.py [hunk 1]
```diff
@@ -62,7 +63,7 @@ class BuildDepCommand(dnf.cli.Command):
 
     def __init__(self, cli):
         super(BuildDepCommand, self).__init__(cli)
-        self.rpm_ts = rpm.TransactionSet()
+        self._rpm_ts = dnf.rpm.transaction.initReadOnlyTransaction()
```
Comment (j-mracek) [inline plugins/builddep.py]: @ignatenkobrain Please what does ```rpm._RPMVSF_NODIGESTS	if set, don't check digest(s).```

The rest looks fine

Comment (ignatenkobrain) [inline plugins/builddep.py]: it disables checking of digests (checksums). According to Florian it's harmless for SRPMs and gives speedup.

**Comment by j-mracek** [comment]
@dnf-bot r+

**Pull request closed without merging**